docs/hooks/hooks.md

```diff
@@ -6,6 +6,28 @@ sidebar_position: 4
 
 Scaffold-Stark 2 provides a collection of custom React hooks designed to simplify interactions with your deployed smart contracts. These hooks are wrappers around Starknet-React, an easy-to-use interface with typescript autocompletions for reading from, writing to, and monitoring events emitted by your smart contracts.
 
+## Using Your Own Contracts
+
+When you want to make use of the contracts you've developed on the `sn-foundry` package, you can run the following command to deploy to `devnet` and generate the ABIs for the hooks to make use of.
+
+```
+yarn deploy
+```
+
+The command will create a file called `packages/nextjs/contracts/deployedContracts.ts` within the Next.js App, which would be the ABI source-of-truth for the Scaffold Hooks. This enables autocomplete to be activated when you are building the Next.js front-end of the application.
+
+:::tip
+
+Note that `yarn deploy` does not reset the the generated `deployedContracts.ts` file, meaning that any changes that you made to the contract name or any deletion will not be taken into account, as it will only make the new deployment on top of the existing API. To completely start from a fresh state of deployment, you can run (do run with caution):
+
+```
+yarn deploy:reset
```

Comment:
couple of points:

1. this is not only for devnet
2. deploy is already on environment tab, we can have this there or point to this one for futther instructions on `reset`

Reply:
will move to Environent tab